src/PIL/PpmImagePlugin.py

```diff
@@ -49,25 +49,39 @@ class PpmImageFile(ImageFile.ImageFile):
     format = "PPM"
     format_description = "Pbmplus image"
 
-    def _token(self, s=b""):
+    def _read_token(self, token=b""):
+        def _ignore_comment():  # ignores rest of the line; stops at CR, LF or EOF
+            while True:
+                c = self.fp.read(1)
+                if c in b"\r\n":
+                    break
+
+        while True:  # read until non-whitespace is found
+            c = self.fp.read(1)
+            if c == b"#":  # found comment, ignore it
+                _ignore_comment()
+                continue
+            if c in B_WHITESPACE:  # found whitespace, ignore it
+                if c == b"":  # reached EOF
+                    raise EOFError("Reached EOF while reading header")
```

Comment:
This looks fine to me, but the exception handler you found in https://github.com/python-pillow/Pillow/discussions/5104#discussioncomment-261675 suggests `EOFError` is only expected in case you "got header but not the first frame". Does anyone think this would be a problem?

Reply:
From what we discussed in that thread I had made my mind to revert to `ValueError` everywhere (except when checking the magic number).

Reply:
This was addressed in https://github.com/python-pillow/Pillow/pull/5121/commits/5d0ad5e2e9a5065d69a00890342334f374277d1c